src/Symfony/Component/PropertyInfo/Util/PhpDocTypeHelper.php

```diff
@@ -113,8 +113,11 @@ private function createType(DocType $type, bool $nullable, string $docType = nul
                 $collectionKeyType = null;
                 $collectionValueType = null;
             } else {
+                // There is no way to declare nullable collection value in PHPDoc syntax, so it is always non-nullable
+                $collectionValueNullable = false;
+
                 $collectionKeyType = new Type(Type::BUILTIN_TYPE_INT);
-                $collectionValueType = $this->createType($type, $nullable, substr($docType, 0, -2));
+                $collectionValueType = $this->createType($type, $collectionValueNullable, substr($docType, 0, -2));
```

Comment:
we should pass `false` here, no need for the extra variable

Reply:
@nicolas-grekas, fixed.